apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -430,19 +432,23 @@ id2bin({FrameworkId, TaskId}) ->
 handle_task_state(TaskObj, _Task) ->
     case maps:get(<<"state">>, TaskObj) of
         TaskState when ?IS_TERMINAL(TaskState) ->
-            false;
+            terminal;
         <<"TASK_RUNNING">> ->
-            Status = handle_task_status(TaskObj),
-            case mget(<<"healthy">>, Status, undefined) of
-                undefined -> running;
-                Healthy ->
-                    % NOTE: it doesn't work, see CORE-1458
-                    {running, Healthy}
-            end;
+            running;
         _TaskState ->
-            true
+            preparing
     end.
 
+-spec(handle_task_healthy(jiffy:object(), task()) -> undefined | boolean()).
+handle_task_healthy(TaskObj, _Task) ->
+    IsHealthy =
+        case maps:is_key(<<"health_check">>, TaskObj) of
+            false -> undefined;
```

Comment:
wouldn't "undefined" break the lists:filter on is_healthy later? We should make it "true" to fall back to the previous behavior.

Reply:
`undefined` wouldn't break any code. In that case `mput/2` doesn't update task info and it doesn't have `healthy` field.